intact-orthology-import/src/main/java/uk/ac/ebi/intact/ortholog/OrthologsFileParser.java

```diff
@@ -0,0 +1,94 @@
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Log4j
+public class OrthologsFileParser {
+
+    public static Map<String, String> parseFile(String filePath) {
+        Pattern uniprotKBRegex = Pattern.compile("UniProtKB=([A-Z0-9]+)");
+        Pattern pantherRegex = Pattern.compile("PTHR\\d+");
+        Map<String, String> uniprotAndPTHR = new HashMap<>();
+        log.info("Parsing file...");
+
+        try (
+                BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+
+            String line;
+            while ((line = reader.readLine()) != null) {
+                ArrayList<String> uniprotMatches = new ArrayList<>();
+
+                Matcher uniprotMatcher = uniprotKBRegex.matcher(line);
+                Matcher pantherMatcher = pantherRegex.matcher(line);
+
+                while (uniprotMatcher.find()) {
+                    uniprotMatches.add(uniprotMatcher.group(1));
+                }
+                while (pantherMatcher.find()) {
+                    uniprotMatches.forEach(up -> uniprotAndPTHR.put(up, pantherMatcher.group()));
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        log.info("File parsed.");
+        log.info("Number of Panther identifiers: " + uniprotAndPTHR.size());
+        return uniprotAndPTHR;
+    }
+
+    public static void parseFileAndSave(String inputFilePath, String outputDirPath) throws IOException {
+        Pattern uniprotKBRegex = Pattern.compile("UniProtKB=([A-Z0-9]+)");
+        Pattern pantherRegex = Pattern.compile("PTHR\\d+");
+        long uniprotAndPantherCount = 0;
+        log.info("Parsing file...");
+
+        File outputDir = new File(outputDirPath);
+        // First, we empty de directory to start clean
+        if (outputDir.exists()) {
+            FileUtils.deleteDirectory(outputDir);
+        }
+        outputDir.mkdirs();
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(inputFilePath))) {
+
+            String line;
+            while ((line = reader.readLine()) != null) {
+                ArrayList<String> uniprotMatches = new ArrayList<>();
+
+                Matcher uniprotMatcher = uniprotKBRegex.matcher(line);
+                Matcher pantherMatcher = pantherRegex.matcher(line);
+
+                while (uniprotMatcher.find()) {
+                    uniprotMatches.add(uniprotMatcher.group(1));
+                }
+                while (pantherMatcher.find()) {
+                    for (String uniprotMatch : uniprotMatches) {
+                        writePair(outputDir.toPath(), uniprotMatch, pantherMatcher.group());
+                        uniprotAndPantherCount += uniprotMatches.size();
+                    }
+                }
+            }
+        }
+
+        log.info("File parsed.");
+        log.info("Number of Panther identifiers: " + uniprotAndPantherCount);
+    }
```

Comment:
We can try using a map again, to avoid duplication, and then saving to files (to be used for later steps).

```suggestion
    public static void parseFileAndSave(String inputFilePath, String outputDirPath) throws IOException {
        Pattern uniprotKBRegex = Pattern.compile("UniProtKB=([A-Z0-9]+)");
        Pattern pantherRegex = Pattern.compile("PTHR\\d+");
        log.info("Parsing file...");

        File outputDir = new File(outputDirPath);
        // First, we empty de directory to start clean
        if (outputDir.exists()) {
            FileUtils.deleteDirectory(outputDir);
        }
        outputDir.mkdirs();

        // First we store all matches in a map to ensure there's no duplication
        // TODO: test this in Codon, as this may cause memory issues
        Map<String, Set<String>> uniprotAndPTHR = new HashMap<>();
        try (BufferedReader reader = new BufferedReader(new FileReader(inputFilePath))) {

            String line;
            while ((line = reader.readLine()) != null) {
                ArrayList<String> uniprotMatches = new ArrayList<>();

                Matcher uniprotMatcher = uniprotKBRegex.matcher(line);
                Matcher pantherMatcher = pantherRegex.matcher(line);

                while (uniprotMatcher.find()) {
                    uniprotMatches.add(uniprotMatcher.group(1));
                }
                while (pantherMatcher.find()) {
                    for (String uniprotMatch : uniprotMatches) {
                        uniprotAndPTHR.putIfAbsent(uniprotMatch, new HashSet<>());
                        uniprotAndPTHR.get(uniprotMatch).add(pantherMatcher.group());
                    }
                }
            }
        }
        
        // Then, we write all the files
        long uniprotAndPantherCount = 0;
        for (String uniprotMatch : uniprotAndPTHR.keySet()) {
            for (String pantherMatch : uniprotAndPTHR.get(uniprotMatch)) {
                writePair(outputDir.toPath(), uniprotMatch, pantherMatch);
            }
            uniprotAndPantherCount += uniprotAndPTHR.get(uniprotMatch).size();
        }

        log.info("File parsed.");
        log.info("Number of Panther identifiers: " + uniprotAndPantherCount);
    }
```